library/Vanilla/Models/ThemeModel.php

```diff
@@ -337,10 +338,10 @@ public function getMasterThemeKey($themeKey): string {
     /**
      * Get view theme addon
      *
-     * @return Addon
+     * @return AddonInterface
      */
-    public function getCurrentThemeAddon(): Addon {
-        $currentTheme = $this->getCurrentTheme();
+    public function getCurrentThemeAddon(): AddonInterface {
```

Comment:
Is this a method of the interface? If not, you should keep the Addon type hint. If yes, the docblock should still list the correct type.

Ideally, this would be the right type for those that don't care about the interface and want to use the full Addon class. Type widening isn't something that's supported right now. You would instead do it without a type hint, but doc blocks only and wait until PHP 7.4 for the type hint.

Reply:
Done